amethyst_rendy/src/submodules/material.rs

```diff
@@ -84,15 +89,24 @@ impl<B: Backend> SlottedBuffer<B> {
         elem_size: u64,
         capacity: usize,
         usage: hal::buffer::Usage,
-    ) -> Result<Self, failure::Error> {
+    ) -> Result<Self, CreationError> {
         Ok(Self {
-            buffer: factory.create_buffer(
-                BufferInfo {
-                    size: elem_size * (capacity as u64),
-                    usage,
-                },
-                rendy::memory::Dynamic,
-            )?,
+            buffer: factory
+                .create_buffer(
+                    BufferInfo {
+                        size: elem_size * (capacity as u64),
+                        usage,
+                    },
+                    rendy::memory::Dynamic,
+                )
+                .map_err(|e| match e {
+                    rendy::resource::CreationError::Allocate(
+                        rendy::memory::HeapsError::AllocationError(
+                            hal::device::AllocationError::OutOfMemory(oom),
+                        ),
+                    ) => oom.into(),
+                    _ => CreationError::Other,
```

Comment:
Given that `rendy` now imposes `CreationError` as the error type for `RenderGroupDesc`, all the error types that were previously `failure::Error` now need to be coerced to it. AFAICT `rendy` doesn't provide any simple way to move between its different error types, so I've manually coerced a few where it was obvious.

Reply:
that should make errors easier to understand in my opinion, nice extra effort to not obscure them!